src/containers/Main/Faucet.tsx

```diff
@@ -92,48 +91,44 @@ const ButtonWrapper = styled.div`
 `;
 
 type HOCProps = FormComponentProps & RouteComponentProps;
-interface FaucetProps extends HOCProps {
-  getFromFaucet: $TSFixMe;
-}
+type FaucetProps = HOCProps;
 
-function Faucet({ form, getFromFaucet }: FaucetProps) {
+function Faucet({ form }: FaucetProps) {
   const { getFieldDecorator } = form;
-  const [isLoading, setIsLoading] = useState(false);
+
+  const {
+    mutate: requestFaucetFunds,
+    isLoading: isRequestFaucetFundsLoading,
+  } = useRequestFaucetFunds({
```

Comment:
If we want pure containers, this should be in a logical component that passes the props `requestFaucetFunds` and `isRequestFaucetFundsLoading` to the presentational component

Reply:
Currently, we are mocking all requests when snapshotting stories. If we don't find the separation valuable we could continue with this strategy.

Reply:
The way I see it, containers can't be pure; their sole purpose is to handle side-effect by passing down props to the component they render. If you're thinking about pages, I'd say they should be separate pure components that compose components together and form the UX using props they're given. Their respective containers would be in charge of handling side-effect (through hooks for the queries for example) and pass down the props needed to the page itself (`requestFaucetFunds` and `isRequestFaucetFundsLoading ` as you mentioned).

Regarding the snapshots, having this separation makes things easier actually. Instead of rendering the containers in Storybook, we can render the page components directly. This way we don't need to mock the requests, we can just pass down mocked properties to the component.

The containers can then be tested functionally.

What do you think of this approach?

Reply:
Sounds good to me, I'll make a separate `Faucet` component that gets imported and rendered by this container